From cbae6c5be1fc57b79711e7c3e8b88609bfbd7823 Mon Sep 17 00:00:00 2001
From: David Drysdale <drysdale@google.com>
Date: Thu, 2 Oct 2014 11:57:19 +0100
Subject: [PATCH 2/9] chaps: Check for D-Bus acquire_name() method

DBus::Connection::acquire_name() does not exist in the upstream version
of libdbus-c++, so check for it and stick to the original request_name()
method if not present.
---
 chaps/Makefile         | 6 ++++++
 chaps/chaps_adaptor.cc | 4 ++++
 2 files changed, 10 insertions(+)

diff --git a/chaps/Makefile b/chaps/Makefile
index e9d024e..7f6ebbb 100644
--- a/chaps/Makefile
+++ b/chaps/Makefile
@@ -130,6 +130,12 @@ DBUS_POLICY = "context=\"default\""
 else
 DBUS_POLICY = "group=\"pkcs11\""
 endif
+DBUS_FLAGS := $(call check_compile_cxx, \
+	'\#include <dbus-c++/dbus.h>\nint main(){\
+	DBus::Connection::SystemBus().acquire_name("dummy");\
+	return 0;}',,-DNO_DBUS_ACQUIRE_NAME)
+CXXFLAGS += $(DBUS_FLAGS)
+
 .PHONY: $(OUT)/org.chromium.Chaps.conf
 $(OUT)/org.chromium.Chaps.conf :
 	sed -e "s/@POLICY_PERMISSIONS@/$(DBUS_POLICY)/" \
diff --git a/chaps/chaps_adaptor.cc b/chaps/chaps_adaptor.cc
index 3e02d24..7f3eb2b 100644
--- a/chaps/chaps_adaptor.cc
+++ b/chaps/chaps_adaptor.cc
@@ -25,7 +25,11 @@ namespace chaps {
 // Helper used when calling the ObjectAdaptor constructor.
 static DBus::Connection& GetConnection() {
   static DBus::Connection connection = DBus::Connection::SystemBus();
+#ifdef NO_DBUS_ACQUIRE_NAME
+  connection.request_name(kChapsServiceName);
+#else
   CHECK(connection.acquire_name(kChapsServiceName));
+#endif
   return connection;
 }
 
-- 
1.9.1

